diff --git a/src/cloudcutter/__main__.py b/src/cloudcutter/__main__.py
index f35b6af..c870c28 100644
--- a/src/cloudcutter/__main__.py
+++ b/src/cloudcutter/__main__.py
@@ -193,14 +193,14 @@ def __configure_local_device_or_update_firmware(args, update_firmware: bool = Fa
     ])
 
     http_server = tornado.httpserver.HTTPServer(application)
-    http_server.listen(80)
+    http_server.listen(80, address=args.ip)
 
     https_server = tornado.httpserver.HTTPServer(application, ssl_options=context)
-    https_server.listen(443)
+    https_server.listen(443, address=args.ip)
 
     # 2018 SDK seems to request that port for some reason
     dns_https_server = tornado.httpserver.HTTPServer(application, ssl_options=context)
-    dns_https_server.listen(4433)
+    dns_https_server.listen(4433, address=args.ip)
 
     tornado.ioloop.IOLoop.current().start()
     sys.exit(0)
