diff --git a/src/setup_apmode.sh b/src/setup_apmode.sh
index 1215684..22921ee 100755
--- a/src/setup_apmode.sh
+++ b/src/setup_apmode.sh
@@ -86,9 +86,12 @@ if [ "${VERBOSE_OUTPUT}" == "true" ]; then
 fi
 dnsmasq --no-resolv --interface=$WLAN --bind-interfaces --listen-address=$GATEWAY --except-interface=lo -K --dhcp-range=$DHCP_RANGE_START,$DHCP_RANGE_END,12h --dhcp-option=option:router,$GATEWAY --dhcp-option=option:dns-server,$GATEWAY,$DNS2,$DNS3 --dhcp-option=option:netmask,255.255.255.0 --dhcp-sequential-ip --address=/#/${GATEWAY} -x $(pwd)/dnsmasq.pid $LOG_OPTIONS
 
-mkdir /run/mosquitto
+mkdir -p /run/mosquitto
 chown mosquitto /run/mosquitto
-echo -e "listener 1883 0.0.0.0\nallow_anonymous true\n" >> /etc/mosquitto/mosquitto.conf
+cat > /etc/mosquitto/conf.d/cloudcutter.conf <<MOSQ_EOF
+listener 1883 0.0.0.0
+allow_anonymous true
+MOSQ_EOF
 /usr/sbin/mosquitto -d -v -c /etc/mosquitto/mosquitto.conf
 
 echo "If your device gets stuck here with no progress after several (at least two) minutes, see https://github.com/tuya-cloudcutter/tuya-cloudcutter/wiki/FAQ#my-device-gets-stuck-after-dhcp-what-can-i-do for additional steps"
